oauthenticator/bitbucket.py

```diff
@@ -36,18 +31,21 @@ def _token_url_default(self):
     def _userdata_url_default(self):
         return "https://api.bitbucket.org/2.0/user"
 
-    team_whitelist = Set(
+    allowed_teams = Set(
         config=True,
         help="""
-        Deprecated, use `BitbucketOAuthenticator.allowed_teams`
+        Allow members of selected Bitbucket teams to sign in.
         """,
     )
 
-    allowed_teams = Set(
+    # _deprecated_oauth_aliases is used by deprecation logic in OAuthenticator
+    _deprecated_oauth_aliases = {
+        "team_whitelist": ("allowed_teams", "0.12.0"),
+        **OAuthenticator._deprecated_oauth_aliases,
+    }
+    team_whitelist = Set(
         config=True,
-        help="""
-        Allow members of selected Bitbucket teams to sign in.
-        """,
+        help="Deprecated, use :attr:`.BitbucketOAuthenticator.allowed_teams`.",
```

Comment:
I've learned that it's a good practice to always have `in VERSION` everywhere you have a 'deprecated' or 'removed' message, so that folks who see it know how to coordinate the deprecations with their version requirements without having to dig around and find the relevant version where things changed.